### DISCUSSION TITLE
Right way to do a join?

### DISCUSSION THREAD

**Discussion by lukehutch** [serverpod/serverpod#1304] [Q&A]
I need to filter the posts a user can see so that posts from any users they have blocked don't show up, and vice versa.

I will have a table `Posts` with a column `authorUserId`, and a table `Blocked` with columns `blockingUserId` and `blockedUserId`. I need to join this to detect if there are any records where `authorUserId == blockingUserId` or `authorUserId == blockedUserId`.

I know that join syntax is being worked on. Is there a way to do this currently in Serverpod, or do I have to wait for the full join system to be completed?

**Comment by SandPod** [comment]
Hey Luke!

We don't have any support in Serverpod currently for one query joins at the moment. So either you will have to create your own raw SQL query to perform the join required or you could solve it in multiple queries and doing some processing on the Dart side.

I have filter on nested relations (https://github.com/serverpod/serverpod/issues/1290) working on a branch right now and hope to get querying working in a day or two, so hopefully this should be available on main by the end of the week.

Note that we are focusing on `one` to `one` relations at the moment and will work on `one` to `many` and `many` to `many` in subsequent steps.

If required there is some join syntax inspiration in the description of the this draft PR (which won't be merged because we have used a different approach): https://github.com/serverpod/serverpod/pull/1268

**Reply by lukehutch** [reply]
Thanks, I knew you were working on something along these lines, but I wasn't sure of the details.

I think what I need is many-to-many, but I'm not sure. Can you please confirm?

Exclusion: I need to filter a query on table A so that the result set includes no records that can be successfully joined with table B, given some join criteria, and some filter criteria in table B.

Inclusion: The same as exclusion, but with inverted logic ("... so that the result set includes only the records that can be ...").

**Reply by SandPod** [reply]
It doesn't seem like one-to-one will enough to cover your case. It sounds like you have two one-to-many relations going on, Users to Posts and Users to Blocked that you would like to filter on.

I notice now that we don't have a separate issue to track filtering on one-to-many or many-to-many. Let me create one and post it here so that you can track it.

**Reply by SandPod** [reply]
Issue: https://github.com/serverpod/serverpod/issues/1307

**Comment by SandPod** [comment]
@lukehutch - This PR, https://github.com/serverpod/serverpod/pull/1399, adds support for filtering a one-to-many relationship. If I understand your problem correctly it should provide the support required to solve your issue.
